sdk/storage/azure-storage-blob-changefeed/src/main/java/com/azure/storage/blob/changefeed/implementation/models/ChangefeedCursor.java

```diff
@@ -0,0 +1,247 @@
+    }
+
+    /**
+     * Creates a new event level cursor.
+     *
+     * @param blockOffset The block offset.
+     * @param objectBlockIndex The object block index.
+     * @return A new event level {@link ChangefeedCursor cursor}.
+     */
+    public ChangefeedCursor toEventCursor(long blockOffset, long objectBlockIndex) {
+        return new ChangefeedCursor(this.getEndTime(), this.getSegmentTime(), this.getShardPath(), this.getChunkPath(), blockOffset, objectBlockIndex);
+
+    }
+
+    /**
+     * @return the end time.
+     */
+    public String getEndTime() {
+        return endTime;
+    }
+
+    /**
+     * @return the segment time.
+     */
+    public String getSegmentTime() {
+        return segmentTime;
+    }
+
+    /**
+     * @return the shard path.
+     */
+    public String getShardPath() {
+        return shardPath;
+    }
+
+    /**
+     * @return the chunk path.
+     */
+    public String getChunkPath() {
+        return chunkPath;
+    }
+
+    /**
+     * @return the block offset
+     */
+    public long getBlockOffset() {
+        return blockOffset;
+    }
+
+    /**
+     * @return the object block index.
+     */
+    public long getObjectBlockIndex() {
+        return objectBlockIndex;
+    }
+
+    /**
+     * @param endTime the end time.
+     * @return the updated BlobChangefeedCursor
+     */
+    public ChangefeedCursor setEndTime(String endTime) {
+        this.endTime = endTime;
+        return this;
+    }
+
+    /**
+     * @param segmentTime the segment time.
+     * @return the updated BlobChangefeedCursor
+     */
+    public ChangefeedCursor setSegmentTime(String segmentTime) {
+        this.segmentTime = segmentTime;
+        return this;
+    }
+
+    /**
+     * @param shardPath the shard path.
+     * @return the updated BlobChangefeedCursor
+     */
+    public ChangefeedCursor setShardPath(String shardPath) {
+        this.shardPath = shardPath;
+        return this;
+    }
+
+    /**
+     * @param chunkPath the chunk path.
+     * @return the updated BlobChangefeedCursor
+     */
+    public ChangefeedCursor setChunkPath(String chunkPath) {
+        this.chunkPath = chunkPath;
+        return this;
+    }
+
+    /**
+     * @param blockOffset the block offset.
+     * @return the updated BlobChangefeedCursor
+     */
+    public ChangefeedCursor setBlockOffset(long blockOffset) {
+        this.blockOffset = blockOffset;
+        return this;
+    }
+
+    /**
+     * @param objectBlockIndex the object block index.
+     * @return the updated BlobChangefeedCursor
+     */
+    public ChangefeedCursor setObjectBlockIndex(long objectBlockIndex) {
+        this.objectBlockIndex = objectBlockIndex;
+        return this;
+    }
+
+    /**
+     * Serializes a {@link ChangefeedCursor} into a String.
+     *
+     * @return The resulting serialized cursor.
+     */
+    public String serialize() {
+        try {
+            return new ObjectMapper().writeValueAsString(this);
```

Comment:
Should make this `ObjectMapper` instance static. For this method should use a `writer()`, for the deserialize case use a `reader()`.

Does the mapper need any configuration to handle empty/null values?

Reply:
No we don't expect any of the values to be null ever when using ser/deser

Reply:
fixed the static comment